### PR TITLE
Add GUI and Lives system to game

### PR DESCRIPTION
**Pull request by LumpBloom7**
This PR adds various GUI elements

It also adds in the lives system along with ghost behaviors after death. (Sorry I worked on too much and its a hassle to separate)

### CODE REVIEW

**Comment by LumpBloom7** [comment]
You may ignore the Codefactor errors

File: Pacman/main.cpp [hunk 1]
```diff
@@ -46,6 +45,34 @@ void test_drawer(const std::filesystem::path& assetsPath)
...
+
+	FontArray.load(FontStyle, 0, { 8 });
+
+	SDL_Renderer* renderer = game->getRenderer();
+
+	bloom::graphics::SpriteText scoreText(renderer, FontArray[0], "Hello World!");
+	bloom::graphics::SpriteText fpsText(renderer, FontArray[0], "Hello World!");
+	bloom::graphics::SpriteText oneUpText(renderer, FontArray[0], "1UP");
+	bloom::graphics::SpriteText hsText(renderer, FontArray[0], "HIGH SCORE");
+
```
Comment (siew24) [inline Pacman/main.cpp]: Maybe a GUISystem can handle this...

Comment (LumpBloom7) [reply Pacman/main.cpp]: Resolved in latest commit

File: Pacman/src/Level.cpp [hunk 2]
```diff
@@ -27,12 +36,28 @@ void Level::draw() {
...
+		m_gameInstance->textures.load(texPath)->render(srcRect, destRect);
+	}
+
 	SDL_SetRenderTarget(m_renderer, nullptr);
 	SDL_RenderCopyEx(m_renderer, m_levelTex, nullptr, &GAMEAREA, 0.0, nullptr, SDL_FLIP_NONE);
 	SDL_RenderCopyEx(m_renderer, m_entityLayer, nullptr, &GAMEAREA, 0.0, nullptr, SDL_FLIP_NONE);
+
+	// Render GUI Text
+	guiElems[0].render(std::nullopt, SDL_Point{ 7 * TILESIZE - guiElems[0].getTextWidth(), 1 * TILESIZE });
+	guiElems[0].render(std::nullopt, SDL_Point{ 17 * TILESIZE - guiElems[0].getTextWidth(), 1 * TILESIZE });
```
Comment (siew24) [inline Pacman/src/Level.cpp]: Intentional?

**Comment by LumpBloom7** [comment]
Yes

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Pull request merged**